integration-tests/modules/__tests__/shipping-options/workflows/update-shipping-options.ts

```diff
@@ -257,6 +257,133 @@ medusaIntegrationTestRunner({
         )
       })
 
+      it("should fail on non-existent region update shipping options", async () => {
+        const regionService = container.resolve(
+          Modules.REGION
+        ) as IRegionModuleService
+
+        const [region] = await regionService.createRegions([
+          {
+            name: "Test region",
+            currency_code: "eur",
+            countries: ["fr"],
+          },
+        ])
+
+        const shippingOptionData: FulfillmentWorkflow.CreateShippingOptionsWorkflowInput =
+          {
+            name: "Test shipping option",
+            price_type: "flat",
+            service_zone_id: serviceZone.id,
+            shipping_profile_id: shippingProfile.id,
+            provider_id,
+            type: {
+              code: "manual-type",
+              label: "Manual Type",
+              description: "Manual Type Description",
+            },
+            prices: [
+              {
+                currency_code: "usd",
+                amount: 10,
+              },
+              {
+                region_id: region.id,
+                amount: 100,
+              },
+              {
+                currency_code: "dkk",
+                amount: 1000,
+              },
+            ],
+            rules: [
+              {
+                attribute: "total",
+                operator: RuleOperator.EQ,
+                value: "100",
+              },
+            ],
+          }
+
+        const { result } = await createShippingOptionsWorkflow(container).run({
+          input: [shippingOptionData],
+        })
+
+        const remoteQuery = container.resolve(
+          ContainerRegistrationKeys.REMOTE_QUERY
+        )
+
+        let remoteQueryObject = remoteQueryObjectFromString({
```

Comment:
Q: nothing to do but should we start using the query graph api instead when writing new things?

Reply:
Yes happy to do that - do you have an example implementation somewhere that I can reference

Reply:
https://github.com/medusajs/medusa/pull/9061
Should give some examples to look at. Let me know how it goes

Reply:
Will probably merge this for now - still happy to do a separate PR with the graph api to try it out though if you can share an example :)